### PR TITLE
salt: fix runtime error on darwin

### LINKED ISSUE #204186: salt package tries finding libcrypto.so instead of libcrypto.dylib on macOS
### Describe the bug

`salt` package tries finding libcrypto.so instead of libcrypto.dylib on macOS.

This results in the following error when trying to run `salt`, `salt-ssh`, etc:

```
  File "/nix/store/$HASH-salt-3005.1/lib/python3.10/site-packages/salt/utils/rsax931.py", line 80, in _find_libcrypto
    raise OSError("Cannot locate OpenSSL libcrypto")
OSError: Cannot locate OpenSSL libcrypto
```

### Steps To Reproduce

1. `nix-env -i salt`
2. `salt`

### Expected behavior

Correct lib is loaded, `salt` works/

### Additional context

When manually changing line `89` in `rsax931.py` to use `.dylib` instead of `.so` it works.

### Notify maintainers

@globin 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.1.0, macOS 13.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.1`
 - channels(chr4): `"nixos-22.11"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Closes #204186.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Flakebi — APPROVED** [review]
Thanks!

**Review by winterqt — APPROVED** [review]

**Comment by winterqt** [comment]
Oh, didn't think of this until just after I hit the merge button, but for the next bump, could a version test be added so we can be sure it doesn't fail to run on any particular platform in the future? Can't hurt, assuming it always tries to load the libraries it needs even if just trying to print the version.

**Pull request merged**
